python/cudf/cudf/core/column/column.py

```diff
@@ -1357,6 +1339,32 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             elif np.issubdtype(data.dtype, np.datetime64):
                 data = data.fillna(np.datetime64("NaT"))
 
+    elif hasattr(arbitrary, "__array_interface__"):
+        # CUDF assumes values are always contiguous
+        desc = arbitrary.__array_interface__
+        shape = desc["shape"]
+        arb_dtype = np.dtype(desc["typestr"])
+        # CUDF assumes values are always contiguous
+        if len(shape) > 1:
+            raise ValueError("Data must be 1-dimensional")
+
+        if desc["strides"] is not None:
+            arbitrary = np.ascontiguousarray(arbitrary)
+        else:
+            arbitrary = np.asarray(arbitrary)
```

Comment:
`np.asarray` can handle non-contiguous inputs as well:
```suggestion
        arbitrary = np.asarray(arbitrary)
        if not arbitrary.flags["C_CONTIGUOUS"]:
            arbitrary = np.ascontiguousarray(arbitrary)
```